packages/ui/src/ui/pages/query-tracker/module/query_result/actions.ts

```diff
@@ -242,3 +250,56 @@ export function updateQueryResult(
         }
     };
 }
+
+export function loadQueryResultsErrors(
+    query: QueryItem,
+): ThunkAction<any, RootState, any, QueryResultsActions> {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const {result_count} = query;
+
+        if (!result_count) return;
+
+        const inds: number[] = [];
+        for (let ind = 0; ind < result_count; ind++) {
+            if (!hasQueryResult(state, query.id, ind)) inds.push(ind);
+        }
+
+        if (inds.length === 0) return;
+
+        try {
+            const {results, error, errorIndices} = await wrapApiPromiseByToaster(
+                dispatch(getQueryResultMetaList(query.id, inds)),
+                {
+                    toasterName: `load_query_results`,
+                    skipSuccessToast: true,
+                    errorTitle: `Failed to load query results meta`,
+                },
+            );
+            forEach_(results, (meta) => {
+                if (meta.error) {
+                    dispatch({
```

Comment:
Each dispatch call enforces to pass an action through all reducers of redux-store, also some react-components might be rerendered for each such call. Probably it is better to collect errors and use one dispatch-call in such cases.

Reply:
Have reworked this part
Now I combine all errors in record, and then call dispatch one time.
Also added new reducer type:  SET_QUERY_RESULTS_ERRORS
for these purpose